worlds/khrecom/Client.py

```diff
@@ -0,0 +1,187 @@
+        global item_num
+        item_num = 1
+
+    @property
+    def endpoints(self):
+        if self.server:
+            return [self.server]
+        else:
+            return []
+
+    async def shutdown(self):
+        await super(KHRECOMContext, self).shutdown()
+        for root, dirs, files in os.walk(self.game_communication_path):
+            for file in files:
+                if file.find("obtain") <= -1:
+                    os.remove(root+"/"+file)
+        global item_num
+        item_num = 1
+
+    def on_package(self, cmd: str, args: dict):
+        if cmd in {"Connected"}:
+            if not os.path.exists(self.game_communication_path):
+                os.makedirs(self.game_communication_path)
+            for ss in self.checked_locations:
+                filename = f"send{ss}"
+                with open(os.path.join(self.game_communication_path, filename), 'w') as f:
+                    f.close()
+            for key in list(args['slot_data'].keys()):
+                with open(os.path.join(self.game_communication_path, key + ".cfg"), 'w') as f:
+                    f.write(str(args['slot_data'][key]))
+                    f.close()
+        if cmd in {"ReceivedItems"}:
+            start_index = args["index"]
+            if start_index != len(self.items_received):
+                global item_num
+                for item in args['items']:
+                    found = False
+                    item_filename = f"AP_{str(item_num)}.item"
+                    for filename in os.listdir(self.game_communication_path):
+                        if filename == item_filename:
+                            found = True
+                    if not found:
+                        with open(os.path.join(self.game_communication_path, item_filename), 'w') as f:
+                            f.write(str(NetworkItem(*item).item) + "\n" + str(NetworkItem(*item).location) + "\n" + str(NetworkItem(*item).player))
+                            f.close()
+                            item_num = item_num + 1
+
+        if cmd in {"RoomUpdate"}:
+            if "checked_locations" in args:
+                for ss in self.checked_locations:
+                    filename = f"send{ss}"
+                    with open(os.path.join(self.game_communication_path, filename), 'w') as f:
+                        f.close()
+
+    def run_gui(self):
+        """Import kivy UI system and start running it as self.ui_task."""
+        from kvui import GameManager
+
+        class KHRECOMManager(GameManager):
+            logging_pairs = [
+                ("Client", "Archipelago")
+            ]
+            base_title = "Archipelago KHRECOM Client"
+
+        self.ui = KHRECOMManager(self)
+        self.ui_task = asyncio.create_task(self.ui.async_run(), name="UI")
+
+
+async def game_watcher(ctx: KHRECOMContext):
+    from .Locations import lookup_id_to_name
+    while not ctx.exit_event.is_set():
+        if ctx.syncing == True:
```

Comment:
Where does `ctx.syncing` get set to true? I'm seeing it get initialized as false and set to false but nothing else. I'm assuming I blind here lol

Reply:
Nice catch, looks like this was leftover from taking code from ChecksFinder.  Its not used in my implementation, so I removed it.